storage/integration_test.go

```diff
@@ -744,6 +746,127 @@ func TestIntegration_ObjectsRangeReader(t *testing.T) {
 	}
 }
 
+func TestIntegration_ObjectReadGRPC(t *testing.T) {
+	// TODO: Remove this once v2 rollout is complete.
+	t.SkipNow()
+
+	ctx := context.Background()
+	client, err := newClientWithGRPC(ctx)
+	if err != nil {
+		t.Error(err)
+	}
+	defer client.Close()
+	bkt := client.Bucket(grpcBucket)
+
+	objName := uidSpace.New()
+	obj := bkt.Object(objName)
+	contents := []byte("Hello, world this is a grpc request")
+
+	if err := retry(ctx, func() error {
+		w := obj.NewWriter(ctx)
+		if _, err := w.Write(contents); err != nil {
+			return fmt.Errorf("Failed to write contents: %v", err)
+		}
+		if err := w.Close(); err != nil {
+			return fmt.Errorf("Failed to close writer: %v", err)
+		}
+		return nil
+	}, nil); err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := obj.Delete(ctx); err != nil {
+			log.Printf("failed to delete test object: %v", err)
+		}
+	}()
+
+	r, err := obj.NewReader(ctx)
+	if err != nil {
+		t.Error(err)
+	}
+	defer r.Close()
+	buf := make([]byte, len(contents))
+	n, err := r.Read(buf)
+	if err != nil && err != io.EOF {
+		t.Error(err)
+		t.FailNow()
+	}
+	if n == 0 && err != io.EOF {
+		t.Error("Expected to have read more than 0 bytes")
+	}
+
+	got := string(buf)
+	want := string(contents)
+	if diff := cmp.Diff(got, want); diff != "" {
+		t.Errorf("got(-),want(+):\n%s", diff)
+	}
+}
+
+func TestIntegration_ObjectReadChunksGRPC(t *testing.T) {
+	// TODO: Remove this once v2 rollout is complete.
+	t.SkipNow()
+
+	ctx := context.Background()
+	client, err := newClientWithGRPC(ctx)
+	if err != nil {
+		t.Error(err)
+	}
+	defer client.Close()
+
+	bkt := client.Bucket(grpcBucket)
+
+	objName := uidSpace.New()
+	obj := bkt.Object(objName)
+	contents := []byte("Hello, world this is a grpc request")
+
+	if err := retry(ctx, func() error {
+		w := obj.NewWriter(ctx)
+		if _, err := w.Write(contents); err != nil {
+			return fmt.Errorf("Failed to write contents: %v", err)
+		}
+		if err := w.Close(); err != nil {
+			return fmt.Errorf("Failed to close writer: %v", err)
+		}
+		return nil
+	}, nil); err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := obj.Delete(ctx); err != nil {
+			log.Printf("failed to delete test object: %v", err)
+		}
+	}()
+
+	r, err := obj.NewReader(ctx)
+	if err != nil {
+		t.Error(err)
+	}
+	defer r.Close()
+
+	bufSize := len(contents)
+	buf := make([]byte, bufSize)
+	chunk := 3
```

Comment:
I see what you're doing here but really I think we need to read a larger file than the message size to verify that the chunking works correctly.

Reply:
Agreed. How would you like to see multi-megabyte test data be generated? Should I just write a bunch of random content in a loop until we hit a good size and then expect that the same amount was read?

Reply:
There are various clever ways to do this in Go. I remember in another test I used [bytes.Repeat](https://pkg.go.dev/bytes#Repeat) to make a larger buffer. Obviously if you want more content than can be held in memory, you need to take a different approach, but this is a simple one anyway.

Reply:
`bytes.Repeat` is super useful. 

> Obviously if you want more content than can be held in memory,

WDYM? I've not done this before so, idk what is acceptable. Could I make a 2+ MB buffer in mem using `bytes.Repeat` to upload then read? Or are you recommending writing 2+ MB worth of data using smaller buffers, then reading it all?